data-analytics/dataflow-python-examples/google/data_enrichment.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" data_enrichment.py demonstrates a Dataflow pipeline which reads a file and
+ writes its contents to a BigQuery table.  Along the way, data from BigQuery
+ is read in as a side input and joined in with the primary data from the file.
+ 
+"""
+
+from __future__ import absolute_import
+import argparse
+
+import logging
+import os
+import apache_beam as beam
+from apache_beam.io.gcp.bigquery import parse_table_schema_from_json
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.pvalue import AsDict
+
+
+class DataIngestion:
+    def __init__(self):
+        """A helper class which contains the logic to translate the file into a
+      format BigQuery will accept"""
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        self.schema_str = ''
+        with open(dir_path + '/resources/usa_names_with_full_state_name.json') \
+                as f:
+            data = f.read()
+            # Wrapping the schema in fields: is required for the BigQuery API
+            self.schema_str = '{"fields": ' + data + '}'
+
+    def parse_method(self, string_input):
+        """This method translates a single line of comma separated values to a
+    dictionary which can be loaded into BigQuery.
+
+    Example Input:
+        KS,F,1923,Dorothy,654,11/28/2016
+
+    Example Output:
+          {'state': 'KS',
+           'gender': 'F',
+           'year': '1923',
+           'name': 'Dorothy',
+           'number': '654',
+           'created_date': '11/28/2016'
+           }
+     """
+        import csv
```

Comment:
This might be due to my own ignorance, but is there a reason for the inline imports?